### ISSUE TITLE
NonNullableFieldWasNullException not handled in the ExecutorServiceExecutionStrategy

### ISSUE THREAD

**Issue by joostlammers** [graphql-java/graphql-java#582]
Whenever a field is null which is mentioned as non-nullable, the ExecutorServiceExecutionStrategy will simply throw an error, instead of bubbling it upwards until either a nullable level or the root is discovered.
This makes it not according to the spec.
[http://facebook.github.io/graphql/#sec-Errors-and-Non-Nullability](http://facebook.github.io/graphql/#sec-Errors-and-Non-Nullability)

To fix this, it is basically nothing more than checking the cause within the ExecutionException and if it is `instanceof NonNullableFieldWasNullException)` then perform similar logic as seen in the SimpleExecutionStrategy.

```
} catch (ExecutionException e) {
    val actualCause = e.getCause();
    if (actualCause != null && actualCause instanceof NonNullableFieldWasNullException){
        val nonNullableFieldWasNullException = (NonNullableFieldWasNullException) actualCause;

        // Following logic is identical to the error handling of the SimpleExecutionStrategy
        // causing the rippling effect downwards for non-nullable fields turning null because of an error
        if (parameters.typeInfo().typeIsNonNull()) {
            throw nonNullableFieldWasNullException;
        }

        return new ExecutionResultImpl(null, executionContext.getErrors());
    }

    throw new GraphQLException(e);
}
```

**Comment by bbakerman** [comment]
I will fix this along with #581

**Comment by bbakerman** [comment]
See https://github.com/graphql-java/graphql-java/pull/591
